ui/src/org/pentaho/di/ui/spoon/Spoon.java

```diff
@@ -6760,20 +6791,24 @@ public void setShellText() {
     String name = null;
     String version = null;
     ChangedFlagInterface changed = null;
+    boolean versioningEnabled = true;
 
+    Object[] metaProperties = null;
     TransMeta transMeta = getActiveTransformation();
-    if ( transMeta != null ) {
-      changed = transMeta;
-      filename = transMeta.getFilename();
-      name = transMeta.getName();
-      version = transMeta.getObjectRevision() == null ? null : transMeta.getObjectRevision().getName();
-    }
     JobMeta jobMeta = getActiveJob();
+
     if ( jobMeta != null ) {
-      changed = jobMeta;
-      filename = jobMeta.getFilename();
-      name = jobMeta.getName();
-      version = jobMeta.getObjectRevision() == null ? null : jobMeta.getObjectRevision().getName();
+      metaProperties = getMetaProperties( rep, jobMeta );
+    } else if ( transMeta != null ) {
```

Comment:
Getting back an array of Objects feels really clunky.  
Is there a reason we can't just do something like

```java
    AbstractMeta meta = Objects.firstNonNull( getActiveJob(), getActiveTransformation() );
    changed = meta;
    filename = meta.getFilename();
    name = meta.getName();
    version = meta.getObjectRevision() == null ? null : meta.getObjectRevision().getName();
    versioningEnabled = isVersioningEnabled( meta );
```

That way we don't need to have to pass around an untyped Object array.